src/util.ts

```diff
@@ -21,126 +17,31 @@ export function noCache(res: http.ServerResponse) {
   res.setHeader('cache-control', 'no-cache, max-age=0, must-revalidate');
 }
 
+export class BasicAuthError extends Error {}
+
 /**
  * sets and parses basic auth headers if they exist
  * @param  req  - http request object
  * @param  res  - http response
  * @param  callback - function(username, password)
  */
-export function basicAuth(
-  req: http.IncomingMessage,
-  res: http.ServerResponse,
-  callback: (username?: string, password?: string) => void
-) {
-  if (!req.headers['authorization']) {
-    res.setHeader('Content-Type', 'text/plain');
-    res.setHeader('WWW-Authenticate', 'Basic realm="authorization needed"');
-    res.writeHead(401);
-    res.end('401 Unauthorized');
-  } else {
+export function parseBasicAuth(
```

Comment:
Can we keep this as basicAuth?

Reply:
yeah sure, I was just trying to be consistent with the function names I'd added in `protocol.ts`